api/pkg/transformer/pipeline/compiler.go

```diff
@@ -198,10 +214,32 @@ func (c *Compiler) parseTablesSpec(tableSpecs []*spec.Table, compiledJsonPaths *
 	return NewCreateTableOp(tableSpecs), nil
 }
 
+func (c *Compiler) parseTableTransform(transformationSpecs *spec.TableTransformation, paths *jsonpath.Storage, compiledExpressions *expression.Storage) (Op, error) {
+	for _, step := range transformationSpecs.Steps {
+		for _, updateColumn := range step.UpdateColumns {
+			compiledExpression, err := c.compileExpression(updateColumn.Expression)
+			if err != nil {
+				return nil, err
+			}
+			compiledExpressions.Set(updateColumn.Expression, compiledExpression)
+		}
+	}
+
+	return NewTableTransformOp(transformationSpecs), nil
+}
+
+func (c *Compiler) parseTableJoin(tableJoinSpecs *spec.TableJoin, paths *jsonpath.Storage, expressions *expression.Storage) (Op, error) {
+	return NewTableJoinOp(tableJoinSpecs), nil
```

Comment:
why you don't check expressions for `leftTable`, `rightTable`?

Reply:
also I think you need to register outputTable variables / dummy table, in case it is referred in the next step of operation

Reply:
Updated